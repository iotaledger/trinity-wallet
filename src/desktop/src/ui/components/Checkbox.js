import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import css from './checkbox.scss';

/**
 * Checkbox component
 */
export default class Checkbox extends React.PureComponent {
    static propTypes = {
        /** Checkbox state */
        checked: PropTypes.bool.isRequired,
        /** Input label */
        label: PropTypes.string,
        /** Address change event function
         * @param {Bool} value - Current checkbox state
         */
        onChange: PropTypes.func.isRequired,
    };

    render() {
        const { checked, label, onChange } = this.props;

        return (
            <div className={classNames(css.checkbox, checked ? css.on : null)} onClick={() => onChange(!checked)}>
                {label}
            </div>
        );
    }
}
