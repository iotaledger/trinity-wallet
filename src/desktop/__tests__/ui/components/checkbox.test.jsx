import React from 'react';
import { shallow } from 'enzyme';

import Checkbox from 'ui/components/Checkbox';

const props = {
    checked: false,
    label: 'Foo',
    onChange: jest.fn(),
};

describe('Checkbox component', () => {
    test('Render the component', () => {
        const wrapper = shallow(<Checkbox {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Unchecked state', () => {
        const wrapper = shallow(<Checkbox {...props} />);

        expect(wrapper.hasClass('on')).toBeFalsy();
    });

    test('Checked state', () => {
        const mockProps = Object.assign({}, props, { checked: true });
        const wrapper = shallow(<Checkbox {...mockProps} />);

        expect(wrapper.hasClass('on')).toBeTruthy();
    });

    test('Checkbox toggle event', () => {
        const wrapper = shallow(<Checkbox {...props} />);

        wrapper.simulate('click');
        expect(props.onChange).toHaveBeenCalledTimes(1);
    });

    test('Checkbox label', () => {
        const wrapper = shallow(<Checkbox {...props} />);

        expect(wrapper.text()).toEqual('Foo');
    });

    test('Disabled state', () => {
        const mockProps = Object.assign({}, props, { disabled: true });
        const wrapper = shallow(<Checkbox {...mockProps} />);

        expect(wrapper.hasClass('disabled')).toBeTruthy();
    });
});
