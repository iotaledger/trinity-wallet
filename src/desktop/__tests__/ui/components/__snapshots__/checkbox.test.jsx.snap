// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component Render the component 1`] = `
<div
  className="checkbox undefined"
  onClick={[Function]}
>
  Foo
</div>
`;
